libs/@local/hash-authorization/src/backend/mod.rs

```diff
@@ -382,7 +437,29 @@ impl CheckResponse {
     }
 }
 
-/// Error returned from [`ZanzibarBackend::check`].
+/// Return value for [`ZanzibarBackend::check_permissions`].
+#[derive(Debug)]
+#[must_use]
+pub struct BulkCheckResponse<I> {
+    /// If the subject has the specified permission or relation to an [`Resource`].
+    pub permission_iterator: I,
```

Comment:
nit: I dislike how this is suffixed with `_iterator`, in that case, it should just `I` should be constrained, that would maybe be a good idea anyway, considering `BulkCheckItem`. (or call it `permissions`)

non-blocking.